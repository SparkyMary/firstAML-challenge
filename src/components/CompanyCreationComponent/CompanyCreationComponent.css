.creationRoot {
  display: flex;
  flex-direction: column;
}
