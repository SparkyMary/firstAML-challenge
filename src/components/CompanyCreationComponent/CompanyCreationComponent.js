import React from 'react'
import PropTypes from 'prop-types'
import TextField from '@material-ui/core/TextField'
import Button from '@material-ui/core/Button'
import './CompanyCreationComponent.css'

const CompanyCreationComponent = ({ name, onChange, onSave }) => {
  return (
    <div className='creationRoot'>
      <TextField
        label='Input name'
        id='cpmpany-name'
        value={name}
        variant='outlined'
        onChange={onChange}
        size='small'
      />

      <Button variant='contained' color='primary' onClick={onSave}>
        Create Company
      </Button>
    </div>
  )
}

CompanyCreationComponent.propTypes = {
  name: PropTypes.string,
  onChange: PropTypes.func,
  onSave: PropTypes.func
}

export default CompanyCreationComponent
