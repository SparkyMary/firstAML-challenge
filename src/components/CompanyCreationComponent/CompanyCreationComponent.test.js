import React from 'react'
import CompanyCreationComponent from './CompanyCreationComponent'
import renderer from 'react-test-renderer'

test('CompanyCreationComponent render', () => {
  const component = renderer.create(
    <CompanyCreationComponent
      name={'AAA'}
      onChange={() => {}}
      onSave={() => {}}
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
