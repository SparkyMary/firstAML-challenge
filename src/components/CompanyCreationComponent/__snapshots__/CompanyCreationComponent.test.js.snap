// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyCreationComponent render 1`] = `
<div
  className="creationRoot"
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
      data-shrink={true}
      htmlFor="cpmpany-name"
      id="cpmpany-name-label"
    >
      Input name
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
        disabled={false}
        id="cpmpany-name"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="AAA"
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-59 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-61 PrivateNotchedOutline-legendNotched-62"
        >
          <span>
            Input name
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Create Company
    </span>
  </button>
</div>
`;
